lib/workspace.rb

```diff
@@ -0,0 +1,52 @@
+require_relative "user"
+require_relative "channel"
+
+class Workspace
+  attr_reader :users, :channels, :selected
+
+  def initialize
+    @users = User.list
+    @channels = Channel.list
+    @selected = nil
+  end
+
+  # modify this
+  def select_channel(user_input)
+    selected = channels.select do |channel|
+      channel.name == user_input || channel.slack_id == user_input
+    end
+    @selected = selected.first
+  end
+
+  def select_user(user_input)
+    @selected = users.select do |user|
+      user.name == user_input || user.slack_id == user_input
+    end
+    @selected = selected.first
+  end
+
+  # used to show details for selected user or channel
+  def show_details(selected)
+    return @selected.details
+  end
```

Comment:
Do you need the parameter here? You're not using it, and your CLI code doesn't expect to supply it.